### ISSUE TITLE
Spack installation and pmi path variable

### ISSUE THREAD

**Issue by Sara-KS** [flux-framework/flux-core-v0.11#23]
We plan to use spack to maintain our installation of flux for the MuMMI project. We would like to advance our software stack to use Spectrum MPI 10.3, but we are aware of the pmi library issue. We do have a version of pmi provided by @dongahn that we can compile. I understand that there is a variable we can set during compilation to specify the pmi path.  However, we are not sure how to direct the spack installation to use this variable to set the correct pmi path. What modifications do we need to make to our spack installation?

**Comment by dongahn** [comment]
@SteVwonder or @twrs: once we talked about packaging up pmi4pmix with Spack. Is the support there?

@Sara-KS: even if we don't have a package for it, I think this is just a matter of setting an environment variable to your externally built pmi4pmix library. I will paste our confluence wiki document once I get to my office.

Regardless, we would first need your Spectrum team to patch their PMIx with the upstream memory bug fix.

**Comment by dongahn** [comment]
FYI --  https://github.com/pmix/pmix/pull/1369

**Comment by SteVwonder** [comment]
> @SteVwonder or @trws: once we talked about packaging up pmi4pmix with Spack. Is the support there?

No.  We threw the idea out there, but never pulled the trigger.  I'll add an issue to the main repo with the other updates that the spack packge needs.  We can add a variant along the lines of `+pmix_compat` to conditionally build and set env vars for pmi4pmix.
